virtualsmartcard/src/vpicc/virtualsmartcard/SmartcardFilesystem.py

```diff
@@ -679,30 +679,43 @@ def _selectFile(self, p1, p2, data):
         P1_PATH_FROM_MF = 0x08
         P1_PATH_FROM_CURRENTDF = 0x09
 
-        if (p1 >> 4) != 0 or p1 == P1_FILE:
+        #if (p1 >> 4) != 0 or 
+        if p1 == P1_FILE:
+            import binascii
+            logging.debug(f"p1 >> 4 or P1_FILE: {binascii.hexlify(data)}")
             # RFU OR
             # When P1='00', the card knows either because of a specific coding
             # of the file identifier or because of the context of execution of
             # the command if the file to select is the MF, a DF or an EF.
-            if data[:2] == inttostring(self.fid):
-                selected = walk(self, data[2:])
-            elif data[:2] == inttostring(self.currentDF().fid):
-                selected = walk(self.currentDF(), data[2:])
-            else:
-                selected = walk(self.currentDF(), data)
+            try:
+                if data[:2] == inttostring(self.fid):
+                    selected = walk(self, data[2:])
+                elif data[:2] == inttostring(self.currentDF().fid):
+                    selected = walk(self.currentDF(), data[2:])
+                else:
+                    selected = walk(self.currentDF(), data)
+            except SwError as e:
+                # If everything fails, look at MF
+                selected = walk(self, data)
```

Comment:
technically speaking, this is an error, because the application specifically did not do the selection from MF; typically, there is no fall back directory, just the error is returned...